pkg/pub_integration/test/search_update_test.dart

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+import 'package:pub_integration/src/fake_pub_server_process.dart';
+import 'package:pub_integration/src/headless_env.dart';
+import 'package:pub_integration/src/pub_puppeteer_helpers.dart';
+import 'package:puppeteer/puppeteer.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('browser', () {
+    late FakePubServerProcess fakePubServerProcess;
+    late final HeadlessEnv headlessEnv;
+    final httpClient = http.Client();
+
+    setUpAll(() async {
+      fakePubServerProcess = await FakePubServerProcess.start();
+      await fakePubServerProcess.started;
+    });
+
+    tearDownAll(() async {
+      await headlessEnv.close();
+      await fakePubServerProcess.kill();
+      httpClient.close();
+    });
+
+    test('bulk tests', () async {
+      final origin = 'http://localhost:${fakePubServerProcess.port}';
+      // init server data
+      await httpClient.post(Uri.parse('$origin/fake-test-profile'),
+          body: json.encode({
+            'testProfile': {
+              'defaultUser': 'admin@pub.dev',
+              'packages': List.generate(
+                100,
+                (i) => {
+                  'name': i % 3 == 0 ? 'flutter_pkg_$i' : 'pkg_$i',
+                  if (i % 11 == 0) 'publisher': 'example.com',
+                },
+              ),
+            },
+          }));
+      await httpClient.get(Uri.parse('$origin/fake-update-search'));
```

Comment:
Shouldn't we make some helper methods in `pub_puppeteer_helpers.dart` for calling things like: `/fake-update-search`

Ideally, such methods could have a documentation comment too, so that we can know what they do.

Example:
```dart

// Why not extend the PubApiClient generated by our code-gen?
class FakePubSiteClient extends PubApiClient {
  final http.Client _client;
  final String _baseUrl;

  FakePubSiteClient(String baseUrl, http.Client client) : _client = client, _baseUrl = baseUrl, super(baseUrl, client);

  /// Trigger update of search...... <what does actually mean>
  Future<void> fakeUpdateSearch() async {...}

  /// Load [profile] into the fake pub site.
  Future<void> loadTestProfile(Object profile) async {....}
}

```

Just an idea....  it could probably be merged with `PubHttpClient`, in fact... why not just use `PubHttpClient.forceSearchUpdate()` in this case here?

Then later we can dream about making `PubHttpClient` a subclass of the auto-generated `PubApiClient`.

Reply:
Maybe a better approach would be to do fake-update-search internally, as part of the test profile load. (updated)